### PR TITLE
Replace constant REMOTE_INSTALL_TIMEOUT with androidInstallTimeout capability

### PR DESCRIPTION
**Pull request by spearminty**
Proposed change in android driver to address appium issue of adb install timeout value hardcoded.
ref: https://github.com/appium/appium-adb/issues/147

My proposal to fix is to add a new capability to appium "--adb-install-timeout"

This PR has updated and passed tests

### CODE REVIEW

File: lib/android-helpers.js [hunk 1]
```diff
@@ -11,7 +11,6 @@ import Bootstrap from 'appium-android-bootstrap';
 import ADB from 'appium-adb';
 
 const REMOTE_TEMP_PATH = "/data/local/tmp";
-const REMOTE_INSTALL_TIMEOUT = 90000; // milliseconds
```
Comment (imurchie) [inline lib/android-helpers.js]: This should remain, to be used as the default (as it stands, there is no default, so if the `adbInstallTimeout` is not given, what happens?).

File: lib/android-helpers.js [hunk 2]
```diff
@@ -215,7 +214,7 @@ helpers.getRemoteApkPath = function (localApkMd5) {
   return remotePath;
 };
 
-helpers.resetApp = async function (adb, localApkPath, pkg, fastReset) {
+helpers.resetApp = async function (adb, localApkPath, pkg, fastReset, adbInstallTimeout) {
```
Comment (imurchie) [inline lib/android-helpers.js]: We should give a default here.

```js
helpers.resetApp = async function (adb, localApkPath, pkg, fastReset, adbInstallTimeout = REMOTE_INSTALL_TIMEOUT) {
```

**Comment by imurchie** [comment]
I like this being configurable. I would suggest it should have a default set to the original static number.

I think it might be better to be something like `androidInstallTimeout` instead of `adbInstallTimeout`, as well, so we don't encode the underlying technology into the name (what happens if/when `adb` is no longer used to install? Does this timeout still apply?).

**Comment by imurchie** [comment]
Otherwise, awesome! Thank you for the work in getting this done!

**Comment by spearminty** [comment]
I agree with your feedback and have made changes accordingly

File: test/functional/android-helper-e2e-specs.js [hunk 3]
```diff
@@ -4,8 +4,12 @@ import helpers from '../../lib/android-helpers';
 import sampleApps from 'sample-apps';
 import ADB from 'appium-adb';
 
-let apiDemos = sampleApps('ApiDemos-debug');
-let appPackage = 'io.appium.android.apis';
+let opts =
```
Comment (jlipps) [inline test/functional/android-helper-e2e-specs.js]: according to our style guide, please make this:

```js
let opts = {
  localApkPath: sampleApps('ApiDemos-debug'),
  pkg: 'io.appium.android.apis',
  androidInstallTimeout: 90000
};
```

(note spacing, line breaks, indentation)

**Comment by jlipps** [comment]
made one comment. thanks @spearminty. I think once you've addressed this and squashed commits this will be good to go

**Comment by spearminty** [comment]
OK- formatting fixed and commits squashed

**Comment by imurchie** [comment]
Excellent! Waiting for build and then I'm happy with merging!

**Comment by imurchie** [comment]
Functional build passed: https://team-appium.ci.cloudbees.com/job/Appium-android-driver/367/console

**Comment by imurchie** [comment]
Thank you for doing this! It's great!

**Pull request merged**